### DISCUSSION TITLE
Tissue-specific System to Cell & Outlier Removal

### DISCUSSION THREAD

**Discussion by basedank** [msraredon/NICHES#19] [Q&A] [answered]
Hello!

I'm currently working through my dataset using NICHES and have a couple of questions:

**System-to-Cell**
I'm trying to see what ligand-receptor interactions define certain cell types. Since this is System-to-Cell, I'm assuming it takes the ligand expression of all cells which falls on the receptor expression of one cell, which is done cell by cell. If this is the case, this would be problematic considering that I have skin biopsies and blood samples in my NICHES object, so the System (ligands) would potentially be comprised of cells that I do not expect to be communicating with the receiving cell in certain situations (ex: purely blood cell being part of the System presenting ligands to purely skin cells). 

- Is my interpretation correct? 

- If so, how could this be remedied (considering that the System.to.Cell assay is created very early, upstream of subsetting certain cell classes, i.e. immune—mesenchyme)?

**Outlier removal:** 
To me, it seems fine if 3-4+ individuals are driving a signaling archetype cluster… but if 1-2 individuals are driving it, I may want to consider excluding those outliers. Sometimes the signaling archetype isn't defined solely by the outlier individual, preventing me from just excluding the entire cluster. The problem is that in this dataset there are ~150 individuals, so it's difficult to understand which patients to exclude by trying to determine which color corresponds to which patient on the UMAP. 

- Not sure if I'm thinking of this in the right way or if there is an easier way of doing this?

Thank you!

**Comment by msraredon** [comment, marked as answer]
This is a great question! I will do my best to answer.

The gist is this: rather than computing SystemToCell on your whole object, break the object by sample or patient and then run SystemToCell on each individual sub dataset. That way you will only ever consider cell crosses / information crosses with cells from within one individual. You then can combine all of those NICHES objects into one big summary NICHES object and process it as normal.  This workflow also allows you to easily identify outlier patients, explore their patient-specific contributions, and decide to either exclude or treat appropriately. Code for this workflow can look like this:

# Split into samples
```
tissue.list <- SplitObject(total.dataset,split.by = 'orig.ident') #modify split.by argument to match desired metadata
```
# Run NICHES by tissue
```
con.list <- list()
for (i in 1:length(tissue.list)){
  print(i)
  Idents(tissue.list[[i]]) <- tissue.list[[i]]$CellType # set this to point to desired metadata slot including your celltype annotations, if they exist
  con.list[[i]] <- RunNICHES(tissue.list[[i]],
                        species = 'rat', #update for a given dataset
                        assay = 'RNA', # Note that we are choosing here to not use imputed data
                        LR.database = 'fantom5', # users choice
                        meta.data.to.map = names(tissue.list[[i]]@meta.data), # this maps all metadata in the source object
                        CellToCell = T,
                        CellToSystem = T,
                        SystemToCell = T,
                        cell_types = 'CellType')
}
```
# Extract System to Cell Measurements, Format, and Save

## Extract NICHES output of interest
```
temp.list <- list()
for(i in 1:length(con.list)){
  temp.list[[i]] <- con.list[[i]]$SystemToCell # Isolate SystemToCell Signaling
  gc()
}
system.to.cell <- merge(temp.list[[1]],temp.list[2:length(temp.list)])
```
## Scale NICHES outputs once merged
```
system.to.cell <- ScaleData(system.to.cell,features = rownames(system.to.cell)) # this is important for looking at all of the data together like this
```
## Save if desired
```
save(system.to.cell,file='system.to.cell.Robj')
```
Make sense?

**Comment by basedank** [comment]
Thanks for the response! That does make sense. So, just to make sure I understand what could be done:

If I want to only take a look at "reasonable" cell-cell crosses (i.e. within the same tissue of origin and, of course, within the same individual), I could split the dataset twice before running NICHES. 1st split level = by patient, 2nd split level = by tissue. So, for each patient we have a skin object and a blood object. Then, I would run NICHES on that Tissue list, combine those resulting NICHES objects into one summary NICHES object, and proceed onwards with the analysis.

Any fundamental flaws in doing so?

**Reply by msraredon** [reply]
First response: yes, this makes sense. 
Second response: in your particular instance, it may make sense to only split the data by pateint and calculate patient-patient crosses including both skin and blood as a set. The reason for this is that the blood is a special category of tissue that does circulate through (all) tissues, and so you may actually be interested in interactions between circulating blood cells and skin-resident parenchymal cells. This is sort of a users-choice situation -- particularly for the SystemToCell and CellToSystem outputs and workflows, you want to think carefully about what you want your "cell system" to be - who to consider as "all talking to each other" for a given analysis. If you are interested in immune interactions though, and in particular immune interactions that can affect skin-resident cells like fibroblasts or endothelial cells or keratinocytes, you are going to want to (probably) include the blood fraction as part of the "tissue" (ie cell system), so that you can see that signal, from, say, circulating monocytes to skin-resident endothelial cells. Clear?
